src/src/com/microsoft/aad/adal/AuthenticationActivity.java

```diff
@@ -1001,10 +1024,15 @@ private void setAccount(final TokenTaskResult result) throws InvalidKeyException
                 // sqllite database. Only Authenticator and similar UID can
                 // access.
                 Gson gson = new Gson();
-                Logger.v(TAG, "app context:" + getApplicationContext().getPackageName()
+                Logger.i(TAG, "app context:" + getApplicationContext().getPackageName()
                         + " context:" + AuthenticationActivity.this.getPackageName()
-                        + " calling packagename:" + getCallingPackage());
+                        + " calling packagename:" + getCallingPackage(), "");
                 StorageHelper cryptoHelper = new StorageHelper(getApplicationContext());
+
+                if (AuthenticationSettings.INSTANCE.getSecretKeyData() == null) {
+                    Logger.i(TAG, "setAccount: user key is null", "");
```

Comment:
Adding logs in general is good, but it is not a good idea to add expensive calls just to add log. Could this be such a case?